ssl/tls_srp.c

```diff
@@ -499,14 +512,16 @@ char *SSL_get_srp_userinfo(SSL *s)
 # define tls1_ctx_ctrl ssl3_ctx_ctrl
 # define tls1_ctx_callback_ctrl ssl3_ctx_callback_ctrl
 
-int SSL_CTX_set_srp_username(SSL_CTX *ctx, char *name)
+int SSL_CTX_set_srp_username(SSL_CTX *ctx, const char *name)
 {
-    return tls1_ctx_ctrl(ctx, SSL_CTRL_SET_TLS_EXT_SRP_USERNAME, 0, name);
+    char* name_tmp = BUF_strdup(name);
+    return tls1_ctx_ctrl(ctx, SSL_CTRL_SET_TLS_EXT_SRP_USERNAME, 0, name_tmp);
```

Comment:
This also leaves the question: who owns name_tmp if ctrl fails. In practice it maybe can't but ownership semantics must be well defined.

This MR needs to be reworked to preserve SSL_ctrl behaviour. You should be able to push the dup into the ctrl I think.